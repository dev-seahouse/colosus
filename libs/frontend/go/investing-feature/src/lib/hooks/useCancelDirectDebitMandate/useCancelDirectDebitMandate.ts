import type {
  InvestorBrokerageDirectDebitMandateCancelRequestDto,
  StandardError,
} from '@bambu/api-client';
import { TransactInvestorAuthenticatedBrokerageApi } from '@bambu/api-client';
import type { UseMutationOptions } from '@tanstack/react-query';
import { useMutation } from '@tanstack/react-query';

export function useCancelDirectDebitMandate(
  queryOptions?: UseMutationOptions<
    Awaited<ReturnType<typeof postToCancelDirectDebitMandate>>,
    StandardError,
    InvestorBrokerageDirectDebitMandateCancelRequestDto
  >
) {
  return useMutation({
    mutationKey: ['cancelDirectDebitMandate'],
    mutationFn: postToCancelDirectDebitMandate,
    ...queryOptions,
  });
}

async function postToCancelDirectDebitMandate(
  args: InvestorBrokerageDirectDebitMandateCancelRequestDto
) {
  const brokerageApi = new TransactInvestorAuthenticatedBrokerageApi();
  const res = await brokerageApi.cancelDirectDebitMandate(args);
  return res.data;
}
export default useCancelDirectDebitMandate;
