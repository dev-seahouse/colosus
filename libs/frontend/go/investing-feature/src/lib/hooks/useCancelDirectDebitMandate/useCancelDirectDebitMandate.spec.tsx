import useCancelDirectDebitMandate from './useCancelDirectDebitMandate';
import {
  act,
  queryClientWrapper,
  renderHook,
  waitFor,
} from '@bambu/react-test-utils';

describe('useCancelDirectDebitMandate', () => {
  const mockOnSuccess = vi.fn();
  afterAll(() => {
    vi.resetAllMocks();
  });
  it('should render successfully', async () => {
    const { result } = renderHook(
      () =>
        useCancelDirectDebitMandate({
          onSuccess: mockOnSuccess,
        }),
      {
        wrapper: queryClientWrapper,
      }
    );
    act(() => result.current.mutate({ mandateId: '123' }));
    await waitFor(() => result.current.isSuccess);
    expect(mockOnSuccess).toHaveBeenCalled();
    expect(result.current.data).toMatchSnapshot();
  });
});
