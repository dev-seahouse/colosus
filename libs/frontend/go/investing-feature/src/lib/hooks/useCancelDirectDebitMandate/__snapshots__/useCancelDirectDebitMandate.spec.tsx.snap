// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useCancelDirectDebitMandate > should render successfully 1`] = `""`;
